api/v1beta1/awscluster_types.go

```diff
@@ -90,6 +90,13 @@ type AWSClusterSpec struct {
 	// IdentityRef is a reference to a identity to be used when reconciling this cluster
 	// +optional
 	IdentityRef *AWSIdentityReference `json:"identityRef,omitempty"`
+
+	// S3Bucket contains options to configure a supporting S3 bucket for this
+	// cluster - currently used for nodes requiring Ignition
+	// (https://coreos.github.io/ignition/) for bootstrapping (requires
+	// BootstrapFormatIgnition feature flag to be enabled).
+	// +optional
+	S3Bucket S3Bucket `json:"s3Bucket,omitempty"`
```

Comment:
A very minor point, should we make this a pointer if its optional?

Reply:
I'm not sure, `Scheme *ClassicELBScheme`, `IdentityRef *AWSIdentityReference`, `ControlPlaneLoadBalancer *AWSLoadBalancerSpec` are pointers, `ControlPlaneEndpoint clusterv1.APIEndpoint`, `AdditionalTags Tags`, `Bastion Bastion` are not.

But by convention, we can make it a pointer.

Reply:
We don't have to change it, and i thnk you make a good point that we need to be consistent. So lets leave it as is...unless you really want to change it.

Reply:
I'm good with leaving it as it is and I'll create an issue for it. And will mark this conversation as resolved.

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/3243